R4ML/tests/testthat/test-ml.model.cox.R

```diff
@@ -40,61 +71,129 @@ test_that("r4ml.coxph", {
-                                 baseline = list("sex_1", "ph_ecog_1", "ph_ecog_2", "ph_ecog_3"))
+                    baseline = list("sex_1", "ph_ecog_1", "ph_ecog_2", "ph_ecog_3"))
 
   expect_true(abs(r4ml_cox_fit@coxModel["age","coef"] - cox_fit$coefficients["age"]) < .1)
   
-  #@TODO test predict
-  
-  #predict(cox_fit, type = "lp")
-  #r4ml_pred <- predict.r4ml.coxph(r4ml_cox_fit, r4ml_lung_pp$data)
-  #predict(cox_fit, type = "expected")
-  #predict(cox_fit, type = "risk", se.fit = TRUE)
-  #predict(cox_fit, type = "terms", se.fit = TRUE)
-  })
+})
 
-test_that("r4ml.coxph accuracy", {
+test_that("r4ml.coxph accuracy model and predict", {
+  
+  # this test case works by building 2 cox models: 1 using survival::coxph and
+  # 1 using R4ML::r4ml.coxph. It then checks to make sure the outputs are
+  # the same for model and predicts.
+  
   df <- survival::lung
-  df$inst <- NULL
-  df$sex <- NULL
-  df <- stats::na.omit(df)
-  colnames(df) <- c("time", "status", "age", "ph_ecog", "ph_karno", "pat_karno",
-                    "meal_cal", "wt_loss")
+  df <- stats::na.omit(df) # omit.na
+  # since r4ml can handle 0 1 but R can handle  1 2 
+  # @TODO, we have to do the auto recoding of the status columns
+  df$status <- df$status - 1
+  colnames(df) <- gsub("\\.", "_", colnames(df)) # change '.' to '_'
 
+  # split the data into two
+  set.seed(1)
+  indexes <- base::sample(1:nrow(df), size = 0.2*nrow(df))
+  train <- df[-indexes, ]
+  test <- df[indexes, ]
+  
+  # since cox models assumes that model and predict have the 
+  # same data set, we only will work with train
+  df <- train
+  
+  # r4ml data.frame
   hf <- as.r4ml.matrix(df)
 
   library("survival")
-  cox_formula <- Surv(time, status) ~ age + ph_karno + pat_karno + wt_loss
 
+  # survival formula
+  cox_formula <- Surv(time, status) ~  age + ph_ecog + ph_karno + pat_karno + wt_loss
+
+  # create the R and R4ML cox's model
   h_cox <- r4ml.coxph(hf, cox_formula)
   r_cox <- coxph(formula = cox_formula, data = df)
 
   h_cm <- h_cox@coxModel
   r_cm <- r_cox$coefficients
 
-  # use this as the R 3.1 version's subpackage testthat 0.9.1 can't handle 
-  # expect_equal. And some of our customer might be using 3.1
+  # compare a few of the coefficients
   expect_true(abs(h_cm["age", "coef"]-r_cm["age"][[1]]) <= 0.01)
-
-  expect_true(abs(h_cm["ph_karno", "coef"]-r_cm["ph_karno"][[1]])<= 0.01)
+  expect_true(abs(h_cm["ph_karno", "coef"]-r_cm["ph_karno"][[1]]) <= 0.01)
 
   expect_true(abs(h_cm["pat_karno", "coef"]-r_cm["pat_karno"][[1]]) <= 0.01)
 
   expect_true(abs(h_cm["wt_loss", "coef"]-r_cm["wt_loss"][[1]]) <= 0.01)
 
+  # test the final prediction
+  
+  # r4ml prediction
   h_predict <- predict(h_cox, data = hf)
-  r_predict <- predict(r_cox, data = df)
+  h_predict_df <- SparkR::as.data.frame(h_predict) # r frame
+  
+  # since R's cox predict need to be called many times,
+  # we have the wrapper for this
+  r_predict <- r_cox_predict(r_cox, df)
```

Comment:
Why did you change `predict(` to `r_cox_predict(`? It seems to have broken this.